examples/keras_rs/dlrm.py

```diff
@@ -0,0 +1,480 @@
+
+    return rmse_value, model.count_params()
+
+
+def print_stats(rmse_list, num_params, model_name):
+    # Report metrics.
+    num_trials = len(rmse_list)
+    avg_rmse = np.mean(rmse_list)
+    std_rmse = np.std(rmse_list)
+
+    if num_trials == 1:
+        print(f"{model_name}: RMSE = {avg_rmse}; #params = {num_params}")
+    else:
+        print(f"{model_name}: RMSE = {avg_rmse} ± {std_rmse}; #params = {num_params}")
+
+
+"""
+## Real-world example
+
+Let's use the MovieLens 100K dataset. This dataset is used to train models to
+predict users' movie ratings, based on user-related features and movie-related
+features.
+
+### Preparing the dataset
+
+The dataset processing steps here are similar to what's given in the
+[basic ranking](/keras_rs/examples/basic_ranking/)
+tutorial. Let's load the dataset, and keep only the useful columns.
+"""
+
+ratings_ds = tfds.load(
+    "movielens/100k-ratings",
+    split="train",
+    download_and_prepare_kwargs={
+        "download_config": tfds.download.DownloadConfig(
+            verify_ssl=False,
+        )
+    },
+)
+
+ratings_ds = ratings_ds.map(
+    lambda x: (
+        {
+            "movie_id": x["movie_id"],
+            "user_id": x["user_id"],
+            "user_gender": tf.cast(x["user_gender"], dtype=tf.int32),
+            "user_zip_code": x["user_zip_code"],
+            "user_occupation_text": x["user_occupation_text"],
+            "raw_user_age": tf.cast(x["raw_user_age"], dtype=tf.float32),
+            "timestamp": tf.cast(x["timestamp"], dtype=tf.float32),
+        },
+        tf.cast(x["user_rating"], dtype=tf.float32),
+    )
+)
+
+"""
+For every categorical feature, let's get the list of unique values, i.e., vocabulary, so
+that we can use that for the embedding layer.
+"""
+
+vocabularies = {}
+for feature_name in (
+    MOVIELENS_CONFIG["categorical_int_features"]
+    + MOVIELENS_CONFIG["categorical_str_features"]
+):
+    vocabulary = ratings_ds.batch(10_000).map(lambda x, y: x[feature_name])
+    vocabularies[feature_name] = np.unique(np.concatenate(list(vocabulary)))
+
+"""
+One thing we need to do is to use `keras.layers.StringLookup` and
+`keras.layers.IntegerLookup` to convert all the categorical features into indices, which
+can then be fed into embedding layers.
+"""
+
+lookup_layers = {}
+lookup_layers.update(
+    {
+        feature: keras.layers.IntegerLookup(vocabulary=vocabularies[feature])
+        for feature in MOVIELENS_CONFIG["categorical_int_features"]
+    }
+)
+lookup_layers.update(
+    {
+        feature: keras.layers.StringLookup(vocabulary=vocabularies[feature])
+        for feature in MOVIELENS_CONFIG["categorical_str_features"]
+    }
+)
+
+"""
+Let's normalize all the continuous features, so that we can use that for the MLP layers.
+"""
+
+normalization_layers = {}
+for feature_name in MOVIELENS_CONFIG["continuous_features"]:
+    normalization_layers[feature_name] = keras.layers.Normalization(axis=-1)
```

Comment:
Sorry to be a pain here, but could you extract the hour of day, hour of week from the timestamp, and map it to a unit circle using sin, cos? That's a better way of encoding timestamp